pkg/controller/cloud/nodecontroller.go

```diff
@@ -155,3 +270,120 @@ func (cnc *CloudNodeController) Run() {
 		}, cnc.nodeMonitorPeriod, wait.NeverStop)
 	}()
 }
+
+func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
+	node := obj.(*v1.Node)
+	instances, ok := cnc.cloud.Instances()
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("cloudprovider does not support instances"))
+		return
+	}
+
+	// This initializes nodes with cloud info
+	// Only initializes nodes that were created with the "ExternalCloudProvider" taint
+	taints := node.Spec.Taints
+
+	var cloudTaint *v1.Taint
+	for _, taint := range taints {
```

Comment:
It would be cool if this would be a separate, unit-testable function, e.g. `GetTaintFromKey()`; it could be used in the above function as well